### PR TITLE
Renames config keys to be linter names

### PR DESCRIPTION
**Pull request by jackvnimble**
This makes it a bit more clear in a config file which linter will
actually used. This change nests the JSHint and ESLint linters
under JavaScript header on the configuration page

https://trello.com/c/CRtnokdz

### CODE REVIEW

File: app/models/config/base.rb [hunk 1]
```diff
@@ -79,7 +79,7 @@ def file_path
     end
 
     def linter_config
-      hound_config.content.slice(*linter_names).values.first
+      hound_config.content.values_at(*linter_names).compact.first
```
Comment (jackvnimble) [inline app/models/config/base.rb]: @gylaz not sure if you still prefer `values_at` when if requires compact (non_extant keys return nil)

**Comment by jackvnimble** [comment]
The aformentioned nesting of ESLint and JSHint under Javascript

image
image

**Comment by jackvnimble** [comment]
Should we update the default config to reflect these changes? Where does that live?

Also, https://raw.githubusercontent.com/houndci/hound/master/config/style_guides/haml.yml is a broken link.

**Comment by thorncp** [comment]
> Should we update the default config to reflect these changes? Where does that live?

I think it makes sense to do that, in an attempt to encourage people to use linter names.

Good catch on the broken link! It should be https://raw.githubusercontent.com/houndci/linters/master/config/haml.yml

File: app/models/config/base.rb (hunk 1 above)
Comment (gylaz) [inline app/models/config/base.rb]: I can go either way. Your call.

**Comment by thorncp** [comment]
Oh, I didn't answer your question. There isn't a default .hound.yml file that gets used. The config/linter objects themselves know if they're enabled by default or not. But a place to update to the linter name keys is on the config page, where it talks about .hound.yml - https://houndci.com/configuration#configuration

File: app/models/config/coffee_script.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ module Config
   class CoffeeScript < Base
```
Comment (gylaz) [inline app/models/config/coffee_script.rb]: What do you think about renaming these files and classes after their linters...

File: app/models/config/coffee_script.rb [hunk 3]
```diff
@@ -2,6 +2,7 @@ module Config
...
     def linter_names
       [
+        "coffeelint",
```
Comment (gylaz) [inline app/models/config/coffee_script.rb]: ... then here we use the old key (`coffeescript`)?

File: app/models/config/swift.rb [hunk 4]
```diff
@@ -1,5 +1,13 @@
 module Config
   class Swift < Base
+    def linter_names
+      [
+        "swiftlint",
+        "swift_lint",
```
Comment (gylaz) [inline app/models/config/swift.rb]: Do we need to support this one?

File: app/views/pages/configuration.haml [hunk 5]
```diff
@@ -10,9 +10,7 @@
       %li
         = link_to "Coffeescript", "#coffeescript"
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: Worth camel-casing this while we're touching this fiel?

**Comment by jackvnimble** [comment]
Ah, cool, I missed that one. Just changed it 👍

File: app/views/pages/configuration.haml [hunk 6]
```diff
@@ -141,92 +139,98 @@
...
+          If you need to change the way Hound is configured, simply copy the
+          #{link_to "default config", javascript_config_url, target: :blank}.
+          into your project, make changes and reference the file in your
+          %em.code .hound.yml
+
+        %p
+          %code.code-block
+            :preserve
+              jshint:
+                config_file: .javascript-style.json
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: How about renaming this to `.jshintrc` which is what JSHint uses by default?

File: app/models/config/coffee_script.rb (hunk 2 above)
Comment (jackvnimble) [inline app/models/config/coffee_script.rb]: I think that's a good idea

File: app/views/pages/configuration.haml [hunk 7]
```diff
@@ -141,92 +139,98 @@
...
+        %p
+          If you would like to ignore certain JavaScript files, simply copy the
+          #{link_to "default config", javascript_ignore_url, target: :blank}
+          into your project, make changes and reference the file in your
+          %em.code .hound.yml
+
+          %code.code-block
+            :preserve
+              jshint:
+                ignore_file: .javascript_ignore
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: `.jshintignore`?

File: app/models/config/swift.rb (hunk 4 above)
Comment (jackvnimble) [inline app/models/config/swift.rb]: Probably not, I was mimicing the java_script one.

File: app/views/pages/configuration.haml (hunk 5 above)
Comment (jackvnimble) [inline app/views/pages/configuration.haml]: Definitely

File: app/views/pages/configuration.haml [hunk 8]
```diff
@@ -246,7 +250,7 @@
       %p
         %code.code-block
           :preserve
-            scss:
+            scss-lint:
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: Is it strange that `coffeelint` doesn't have a hyphen but this does? Worth standardizing on one way?

File: app/views/pages/configuration.haml (hunk 6 above)
Comment (jackvnimble) [inline app/views/pages/configuration.haml]: Good idea.

File: app/models/config/base.rb (hunk 1 above)
Comment (jackvnimble) [inline app/models/config/base.rb]: I think I'll keep it. `values_at` is much more clear than slice

File: app/views/pages/configuration.haml (hunk 8 above)
Comment (thorncp) [inline app/views/pages/configuration.haml]: The CLI commands are `coffeelint` and `scss-lint`. Makes sense to me for our config keys to be consistent with that.

File: app/views/pages/configuration.haml [hunk 9]
```diff
@@ -299,7 +303,7 @@
       %p
         %code.code-block
           :preserve
-            haml:
+            haml_lint:
```
Comment (gylaz) [inline app/views/pages/configuration.haml]: This one is `_`, but the one above is `-`.

File: app/views/pages/configuration.haml (hunk 8 above)
Comment (gylaz) [inline app/views/pages/configuration.haml]: Maybe we should support both? If someone configures `coffeelint` first, they are likely to assume SCSS will be `scsslint`, and vice versa. It's just one extra entry in the `linter_names` array. Or am I being paranoid?

File: app/views/pages/configuration.haml (hunk 9 above)
Comment (jackvnimble) [inline app/views/pages/configuration.haml]: Good catch, should be `-`

File: app/views/pages/configuration.haml (hunk 8 above)
Comment (thorncp) [inline app/views/pages/configuration.haml]: I like supporting both. More flexibility will probably make for a better user experience.

Comment (jackvnimble) [inline app/views/pages/configuration.haml]: So add `coffee-lint` and `scsslint`?

Comment (gylaz) [inline app/views/pages/configuration.haml]: Yes to `scsslint`.

`coffee-lint`... maybe?

File: spec/models/linter/rubocop_spec.rb [hunk 10]
```diff
@@ -697,19 +697,19 @@ def violations_with_config(config = stub_ruby_config)
 
     def build_linter(
       hound_config: build_hound_config,
-      config: stub_ruby_config,
+      config: stub_rubocop_config,
```
Comment (jackvnimble) [inline spec/models/linter/rubocop_spec.rb]: Yeah... this took me a while to figure out :(

Comment (teoljungberg) [inline spec/models/linter/rubocop_spec.rb]: Yeah, I'm not happy on how this turned out either - saying this as the guy who wrote it.

**Comment by gylaz** [comment]
Closing this due to inactivity. We still want to do this, but probably handle this one linter at a time.

**Pull request closed without merging**